backend/dal.keycloak/Partials/PimsKeycloakUserService.cs

```diff
@@ -160,6 +162,51 @@ public async Task<Entity.User> UpdateUserAsync(Entity.User user)
 
             return euser;
         }
+
+        /// <summary>
+        /// Updates the specified access request in the datasource. if the request is granted, update the associated user as well.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public async Task<Entity.AccessRequest> UpdateAccessRequestAsync(Entity.AccessRequest entity)
+        {
+            entity.ThrowIfNull(nameof(entity));
+            entity.ThrowIfNull(nameof(entity.UserId));
+
+            this._user.ThrowIfNotAuthorized(Permissions.AdminUsers, Permissions.AgencyAdmin);
+            var accessRequest = _pimsAdminService.User.GetAccessRequestNoTracking(entity.Id);
+            if (!accessRequest.IsGranted != true && entity.IsGranted == true)
```

Comment:
/nit: Please clean up the logic here. Double negations are harder to understand.

Isn't `if (!accessRequest.IsGranted != true && entity.IsGranted == true)` the same as `if (accessRequest.IsGranted == true && entity.IsGranted == true)` ?

If the property can be null I'd rather do `var value = accessRequest.IsGranted ?? false` and then check the value?

Reply:
should actually be accessRequest.IsGranted != true, not sure how this works as is. Will update in my next PR.